website_product/controllers/main.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import http
+from odoo.http import request
+
+
+class WebsiteProductPage(http.Controller):
+
+    @http.route(['/product/<model("product.template"):product>'], type='http',
+                auth="public", website=True)
+    def products_detail(self, product, **post):
+        if product.exists:
+            if product.website_published or self._is_website_publisher():
```

Comment:
edit_page is for? Judging from the name I assume it should be there if `product.website_published` am I wrong?